ra-optimisation/search-tree-rao/src/test/java/com.farao_community.farao.search_tree_rao/process/search_tree/TreeTest.java

```diff
@@ -32,58 +48,72 @@
 
         // build output
-        RaoComputationResult result = Tree.buildOutput(leafRoot, leafOptimal);
-
-        assertEquals(2, result.getPreContingencyResult().getMonitoredBranchResults().size());
-        assertEquals("MONITORED_BRANCH_1", result.getPreContingencyResult().getMonitoredBranchResults().get(0).getId());
-        assertEquals(105.0, result.getPreContingencyResult().getMonitoredBranchResults().get(0).getPreOptimisationFlow(), DOUBLE_TOLERANCE);
-        assertEquals(95.0, result.getPreContingencyResult().getMonitoredBranchResults().get(0).getPostOptimisationFlow(), DOUBLE_TOLERANCE);
-
-        assertEquals(2, result.getContingencyResults().size());
-        assertEquals("CONTINGENCY_1", result.getContingencyResults().get(0).getId());
-        assertEquals("MONITORED_BRANCH_1_CO_1", result.getContingencyResults().get(0).getMonitoredBranchResults().get(0).getId());
-        assertEquals(115.0, result.getContingencyResults().get(0).getMonitoredBranchResults().get(0).getPreOptimisationFlow(), DOUBLE_TOLERANCE);
-        assertEquals(98.0, result.getContingencyResults().get(0).getMonitoredBranchResults().get(0).getPostOptimisationFlow(), DOUBLE_TOLERANCE);
-
-        assertEquals(2, result.getPreContingencyResult().getRemedialActionResults().size());
-        assertEquals("RA1", result.getPreContingencyResult().getRemedialActionResults().get(0).getId());
-        assertEquals("PRA_PST_BE", result.getPreContingencyResult().getRemedialActionResults().get(1).getId());
-
-        assertNotNull(result.getExtension(SearchTreeRaoResult.class));
-        assertEquals(SearchTreeRaoResult.ComputationStatus.SECURE, result.getExtension(SearchTreeRaoResult.class).getComputationStatus());
-        assertEquals(SearchTreeRaoResult.StopCriterion.OPTIMIZATION_FINISHED, result.getExtension(SearchTreeRaoResult.class).getStopCriterion());
+        RaoResult result = Tree.buildOutput(rootLeaf, optimalLeaf);
+
+        assertTrue(result.isSuccessful());
+        assertEquals("rootPreOptim", result.getPreOptimVariantId());
+        assertEquals("leafPostOptim", result.getPostOptimVariantId());
     }
 
     @Test
     public void brokenRootSearchTest() {
         Network network = Mockito.mock(Network.class);
         VariantManager variantManager = Mockito.mock(VariantManager.class);
         Mockito.when(network.getVariantManager()).thenReturn(variantManager);
-        RaoComputationResult result;
+        RaoResult result;
         try {
             result = Tree.search(network, Mockito.mock(Crac.class), "", Mockito.mock(RaoParameters.class)).get();
-            assertEquals(RaoComputationResult.Status.FAILURE, result.getStatus());
-            assertEquals(SearchTreeRaoResult.ComputationStatus.ERROR, result.getExtension(SearchTreeRaoResult.class).getComputationStatus());
+            assertEquals(RaoResult.Status.FAILURE, result.getStatus());
         } catch (Exception e) {
             throw new AssertionError();
         }
```

Comment:
I don't understand this try catch

Reply:
The Tree.search method returns a completableFuture. So the .get() can cause an Exception.

I removed the try/catch and added the throws to the method signature.